docs/developer-docs/integrations/sns/get-sns/local-testing.md

```diff
@@ -1,74 +1,150 @@
+:::
+
 #### 1. Prepare your principals and tools
 
-<!-- TODO-content & TODO-CLI/dfx: 
-Open terminal with dfx, ready for commands when we say "use `dfx` identity 
-`identityDevDfx`.
-For things to do with `sns-quill` will say "use `sns-quill` principal `identityDevNeuron`"
-For production, we recommend that message are signed on air-gapped computer and
-sent to IC on connected computer. For testing this does not matter that much.
-But in that case, we recommend that you use a different principal for testing and
-in production (to protect the relevant key).
+To deploy and test a local SNS, you will need the following tools:
+
+- [dfx](../../../build/install-upgrade-remove.mdx)
+- [sns-quill](https://github.com/dfinity/sns-quill)
+
+To install an NNS locally (which is needed to obtain an SNS in the same way that you would get it in production), you have to run your local replica as a
+node of a `system` subnet (as opposed to `application`, which is what all other dapps run on).
+To set your local replica to subnet type `system`, add the following definition to your `networks.json`:
+
+``` json
+{
+  "local": {
+      "bind": "127.0.0.1:8080",
+      "type": "ephemeral",
+      "replica": {
+        "subnet_type": "system"
+      }
+    }
+}
+```
+
+If you are unsure where `networks.json` is located, `dfx info networks-json-path` will tell you.
 
-Might want to set up fake principals for some testing rounds as you want to reach majority in
-voting decisions.
+To test the whole process properly, you also need a handful of identities.
+For local testing, we recommend you use unencrypted/plaintext .pem files to make everything a little bit easier.
+To do so, use the `--disable-encryption` flag when importing/creating an identity in dfx.
 
-Set up NNS in such a way that can vote through proposals. Maybe one huge neuron `identityMajority`?
-Have principal w/ NNS neuron NNS identity `identityNNS`.
--->
+The most important identity you will use is the identity we call `developer-identity`.
+Make sure that you can use it both with `dfx` and `sns-quill`, and that it will receive an initial developer neuron with a majority stake in the initial SNS parameters.
+If it doesn't receive a majority stake, you will not be able to test the upgrade proposals during the decentralization sale.
 
-#### 2. Install an NNS on your local testing environment.
+`dfx nns install` is currently hard-coded to give a certain account ICP to test with tokens.
+Because of that, make sure you also [import](../../../../references/cli-reference/dfx-identity.md#dfx-identity-import) the following identity so you get access to those funds:
+
+```
+-----BEGIN EC PRIVATE KEY-----
+MHQCAQEEICJxApEbuZznKFpV+VKACRK30i6+7u5Z13/DOl18cIC+oAcGBSuBBAAK
+oUQDQgAEPas6Iag4TUx+Uop+3NhE6s3FlayFtbwdhRVjvOar0kPTfE/N8N6btRnd
+74ly5xXEBNSXiENyxhEuzOZrIWMCNQ==
+-----END EC PRIVATE KEY-----
+```
+
+This identity will be referred to as the `nns-ledger-default-identity` for the rest of this guide.
+
```

Comment:
Here I asked myself : at which step was the NNS created? 
Maybe we can make it more prominent in 2. (which I assume is this step) that "this creates a NNS locally" so that a user know when to expect the NNS canisters to be there?